classifier_load_testing/create_groups.rb

```diff
@@ -19,52 +19,51 @@
   File.write('timing.csv',"node, group, parent, elapsedTime \n")
 end
 
-overallStartTime = Time.now
-
-num_nodes.to_i.times do |node|
-  node = node + start_at
-  node_name = "node#{node}"
-  if (node % 100) == 0
-    puts(node_name)
+overallElapsedTime = time_block do
+  # Common helper that removes the "with_retry", "time_block" clutter.
+  # Helper returns how long it took to create the group.
+  create_group = lambda do |group|
+    retryDelay = per_node_delay.zero? ? 5 : 5 * per_node_delay
```

Comment:
`3` is an arbitrary # of retries. retryDelay is also arbitrary. Figured 5 * node_delay should give API more time to settle down.

Reply:
👍